### DISCUSSION TITLE
What is the recommended way to access resources strings programmatically when the resw file is in a Library project (as in the latest templates)?

### DISCUSSION THREAD

**Discussion by BrianDT** [unoplatform/uno#11658] [Q&A] [answered]
I can see that there are solutions for x:Uid localisation confirmed in issue #11462
However programmatic loading of resources only seems to work in net7.0-windows10.0.19041.0 WinUI when 
Something like 

var resourceLoader = ResourceLoader.GetForViewIndependentUse(assemblyName + "/Resources");

Is used

This is not available for WASM, IOS,ANDROID and using GetForCurrentView(); either with or without a assembly name does not enable the return of any values when GetString(key) is called on the resultant resourceLoader.

**Comment by jeromelaban** [comment]
Thanks for the question. Can you specify which version of Uno you are using?

**Reply by BrianDT** [reply]
I am using 4.7.44 Uno.WinUI in the net7.0 net7.0-android and net7.0-ios environments.
The net7.0-windows10.0.19041.0 WinUI head which works has all Uno refs stripped out.

**Reply by jeromelaban** [reply]
Thanks. Are you able to provide a sample? We'll track it in an issue.

**Reply by BrianDT** [reply]
It would seem that I did not retest all the options after implementing the solution to #11462
I can now confirm the functionality is as expected when using the lines
#if HAS_UNO
                    this.resourceLoader = ResourceLoader.GetForCurrentView(assemblyName + "/Resources");
#else
                    this.resourceLoader = ResourceLoader.GetForViewIndependentUse(assemblyName + "/Resources");
#endif

This has been validated in the sample located at https://github.com/BrianDT/UnoResourceStringsIssue.git
The sample is currently set up to test the alternative French resw, override lines in App.cs can be commented out to use the default.

**Reply by jeromelaban** [reply]
Thanks. I'm surprised that you need both GetForCurrentView and GetForViewIndependentUse, as Uno does not make a difference between those calls: 
https://github.com/unoplatform/uno/blob/560a44980945eb8343025832fe0ea23941c99435/src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs#L114-L118

Indeed the missing PRI inclusion is the cause for the missing resources, though.

**Reply by BrianDT** [reply]
You are right, I had not realised, I changed GetForCurrentView to GetForViewIndependentUse for the non Uno case because WinUI 3 raised an error when using GetForCurrentView. Did not consider changing it for Uno as well.

**Reply by jeromelaban** [reply]
Thanks for the update!

**Comment by BrianDT** [comment]
No code issue needs to be raised, but a documentation update would help.

**Comment by BrianDT** [comment, marked as answer]
The combination of the following seems to be the answer

In the library project add
	```<ItemGroup>
		<PRIResource Include="**\*.resw" />
	</ItemGroup>```

Use Uno >= 4.7.44

Create the resource loader using

this.resourceLoader = ResourceLoader.GetForViewIndependentUse(assemblyName + "/Resources");
